unstructured/ingest/v2/processes/connectors/opensearch.py

```diff
@@ -0,0 +1,82 @@
+import json
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Optional
+
+from unstructured.ingest.enhanced_dataclass import enhanced_field
+from unstructured.ingest.error import DestinationConnectionError
+from unstructured.ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+)
+from unstructured.ingest.v2.processes.connector_registry import (
+    DestinationRegistryEntry,
+    add_destination_entry,
+)
+from unstructured.ingest.v2.processes.connectors.elasticsearch import (
+    ElasticsearchUploader,
+    ElasticsearchUploaderConfig,
+    ElasticsearchUploadStager,
+    ElasticsearchUploadStagerConfig,
+)
+from unstructured.utils import requires_dependencies
+
+if TYPE_CHECKING:
+    from opensearchpy import OpenSearch
+
+CONNECTOR_TYPE = "opensearch"
+
+"""Since the actual OpenSearch project is a fork of Elasticsearch, we are relying
+heavily on the Elasticsearch connector code, inheriting the functionality as much as possible."""
+
+
+@dataclass
+class OpenSearchAccessConfig(AccessConfig):
+    password: Optional[str] = enhanced_field(default=None, sensitive=True)
+    use_ssl: bool = False
+    verify_certs: bool = False
+    ssl_show_warn: bool = False
+    ca_certs: Optional[str] = None
+    client_cert: Optional[str] = None
+    client_key: Optional[str] = None
+
+
+@dataclass
+class OpenSearchConnectionConfig(ConnectionConfig):
+    hosts: Optional[list[str]] = None
+    username: Optional[str] = None
+    access_config: OpenSearchAccessConfig = enhanced_field(sensitive=True)
+
+    def to_dict(self, **kwargs) -> dict[str, json]:
```

Comment:
To keep things consistent, can we use similar logic to what's done in the elastisearch connector configs for this: [elasticsearch.py#L72](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/ingest/v2/processes/connectors/elasticsearch.py#L72)

Reply:
Definitely. But its so messy. Whats the advantage? Is Elasticsearch more picky on what is in the dictionary?

Reply:
Allows us to strictly type to expected value to match what's expected by the client and also log exactly what parameters are being passed in with support for sensitive fields.